src/tests/ftest/util/ec_utils.py

```diff
@@ -406,63 +407,43 @@ def start_online_single_operation(self, operation, parity=1):
         while not self.out_queue.empty():
-            if self.out_queue.get() == "FAIL":
+            if (self.out_queue.get() == "FAIL" or self.out_queue.get() == "ERR"):
                 self.fail("FAIL")
 
 
 class ErasureCodeMdtest(MdtestBase):
     """Class to used for EC testing for MDtest Benchmark."""
 
-    def __init__(self, *args, **kwargs):
-        """Initialize a MdtestBase object."""
-        super().__init__(*args, **kwargs)
-        self.server_count = None
-        self.set_online_rebuild = False
-        self.rank_to_kill = None
-        self.obj_class = None
-
     def setUp(self):
         """Set up each test case."""
         super().setUp()
-        engine_count = self.server_managers[0].get_config_value("engines_per_host")
-        self.server_count = len(self.hostlist_servers) * engine_count
-        self.obj_class = self.params.get("dfs_oclass_list", '/run/mdtest/objectclass/*')
         # Create Pool
         self.add_pool()
         self.out_queue = queue.Queue()
 
-    def write_single_mdtest_dataset(self):
-        """Run MDtest with EC object type."""
-        # Update the MDtest obj class
-        self.mdtest_cmd.dfs_oclass.update(self.obj_class)
+    def start_online_mdtest(self, ranks_to_stop):
+        """Run mdtest and stop ranks while mdtest is running.
 
-        # Write the MDtest data
-        self.execute_mdtest(self.out_queue)
-
-    def start_online_mdtest(self):
-        """Run MDtest operation with thread in background.
-
-        Trigger the server failure while MDtest is running
+        Args:
+            ranks_to_stop (list): ranks to stop while mdtest is running
         """
         # Create the MDtest run thread
-        job = threading.Thread(target=self.write_single_mdtest_dataset)
+        job = threading.Thread(
+            target=self.execute_mdtest,
+            kwargs={"out_queue": self.out_queue})
 
         # Launch the MDtest thread
         job.start()
 
-        # Kill the server rank while IO operation in progress
-        if self.set_online_rebuild:
-            time.sleep(30)
-            # Kill the server rank
-            if self.rank_to_kill is not None:
-                self.server_managers[0].stop_ranks([self.rank_to_kill],
-                                                   self.d_log,
-                                                   force=True)
+        # Stop the server ranks while IO operation in progress
+        time.sleep(self.mdtest_cmd.stonewall_timer.value / 2)
+        self.server_managers[0].stop_ranks(ranks_to_stop, self.d_log, force=True)
 
         # Wait to finish the thread
         job.join()
 
         # Verify the queue result and make sure test has no failure
         while not self.out_queue.empty():
-            if self.out_queue.get() == "Mdtest Failed":
-                self.fail("FAIL")
+            result = self.out_queue.get()
+            if result == "Mdtest Failed":
+                self.fail(result)
```

Comment:
The problem is this is only in the queue if mdtest raises `CommandFailure`. Any other exception would be ignored